src/Scene.ts

```diff
@@ -682,18 +683,114 @@ export class Scene {
     this.COMvisual.add(mesh);
   }
 
-  public addSky(): void {
-    var cubeLoader = new THREE.CubeTextureLoader();
-    var cubeTexture = cubeLoader.load([
-      'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-negx.jpg',
-      'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-posx.jpg',
-      'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-posy.jpg',
-      'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-negy.jpg',
-      'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-negz.jpg',
-      'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-posz.jpg',
-    ]);
-
-    this.scene.background = cubeTexture;
+  public addSky(cubemap: string | undefined): void {
+    if (cubemap === undefined) {
+      var cubeLoader = new THREE.CubeTextureLoader();
+      this.scene.background = cubeLoader.load([
+        'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-negx.jpg',
+        'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-posx.jpg',
+        'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-posy.jpg',
+        'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-negy.jpg',
+        'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-negz.jpg',
+        'https://fuel.gazebosim.org/1.0/openrobotics/models/skybox/tip/files/materials/textures/skybox-posz.jpg',
+      ]);
+    } else {
+      let ddsLoader = new DDSLoader();
```

Comment:
nit: const over let

```suggestion
      const ddsLoader = new DDSLoader();
```

Reply:
Since we do check onError if the asset came from the websocket, pass the websocket loading manager into it.

You can check on other loaders how it was used.

Reply:
I think I did it in [eab4493](https://github.com/gazebo-web/gzweb/pull/42/commits/eab44938f9b4cb14ce5e76e9316e5a3cceee1b45)